content/tutorials/portenta-h7/por-ard-flash/content.md

```diff
@@ -0,0 +1,256 @@
+-   (How) to calculate the size of the flash memory 
+
+## Required Hardware and Software
+-   Portenta H7 board (<https://store.arduino.cc/portenta-h7>)
+-   USB C cable (either USB A to USB C or USB C to USB C)
+-   Arduino IDE 1.8.10+ or Arduino Pro IDE 0.0.4+ 
+
+
+# Configuring the Development Environment
+The H7 comes with 2 Mbytes of Flash memory with read-while-write support + 1 Mbyte of RAM. To support the use of non-erasable memory, a part of the flash memory can be programmed to behave as a block device, that can operate as a place to store information. 
+
+To access the internal flash of the MCU, you will have to use the Flash In-Application Programming Interface (FlashIAP). This API creates a block device on top of the space still available on the flash after flashing your sketch. Please**,** note that **the free space** available for the FlashIAP block device **depends on** the space occupied by **the complete sketch** on the flash and a few **block alignments calculations**. You can then use a raw access API to save and load data from the block device.
+
+https://os.mbed.com/docs/mbed-os/v6.4/apis/flash-iap.html
+
+Please, be aware of the **flash r/w limits** while using raw/direct access: flash memories have a limited amount of write cycles. Typical flash memories can perform about 10000 writes cycles to the same block before starting to "wear out" and begin to lose the ability to retain data. **You can actually render your board useless with improper use of this example and described APIs.**
+
+[note]
+
+Note: **Limit** the usage of FlashIAP block devices to **once-in-a-time** read and write **operations**, for example, for managing `setup()`-time configuration parameters.
+**Please clarify. How do I do that? Lead with action. ** 
+
+[/note]
+
+## 1. The Basic Setup
+
+Begin by plugging in your Portenta board to the computer using a USB-C  cable and open the Arduino IDE or the Arduino Pro IDE. If this is your first time running Arduino sketch files on the board, we suggest you  check out how to [set up the Portenta H7 for Arduino](por-ard-gs) before you proceed.
+
+![The Portenta H7 can be connected to the computer using an appropriate USB-C cable](assets/por_ard_usbh_basic_setup.svg)
+
+## 2. Create the FlashStorage_MbedOS.ino
+
+Lets start by creating a sketch named `FlashStorage_MbedOS.ino` file 
+
+Add the FlashIAP support with the relevant library:
+
+```
+#include <FlashIAPBlockDevice.h>
+```
+
+Copy and paste the following code inside `void setup()` 
+
+```cpp
+void setup()
+{
+
+Serial.begin(115200);
+  while (!Serial)
+    ;
+
+//  Wait for terminal to come up
+  delay(1000);
+
+Serial.println("FlashIAPBlockDevice Test");
+
+// Feed the RNG for later random content generation
+  srand(micros());
+
+}
+```
+
+## 3. Define the FlashIAPLimits.h 
+
+These helper functions will help you calculating limits for the FlashIAP block device. You will find them in the header file called FlashIAPLimits.h
```

Comment:
```suggestion
These helper functions will help you calculating limits for the FlashIAP block device. You will find them in the header file called FlashIAPLimits.h
```
These helper functions will help you calculating limits for the FlashIAP block device. You will find them in the header file called `FlashIAPLimits.h`